src/APU.cpp

```diff
@@ -373,26 +357,16 @@ void APU::updateChannel1() {
 		uint8_t waveRatio = squareWaveRatio[wavePatternDutyIndex];
 
 		// Update the sample produced in accordance with the wave frequency
-		channel1.sample = (channel1.waveIndex & 0x07) < waveRatio ? -1.0f : 1.0f;
+		channel1.sample = (channel1.waveIndex & 0x07) < waveRatio ? 0 : 16;
 
 		// Letting it just wrap around for now
 		channel1.waveIndex++;
 	}
 	
-	// Fill the buffer every tick
-	channel1.buffer[channel1.bufferIndex] = channel1.soundOn * channel1.volume * channel1.sample;
```

Comment:
I originally had a buffer because I had plans to implement a low pass filter, and I would have needed some of the previous samples to calculate that. It's something I could maybe return to in the future, but for now in the interest of performance I'm only storing the current sample.